subprojects/execution/src/main/java/org/gradle/internal/execution/impl/OutputsCleaner.java

```diff
@@ -58,6 +64,9 @@ private void cleanupOutput(String absolutePath, FileType fileType) throws IOExce
                 File file = new File(absolutePath);
                 if (fileSafeToDelete.test(file)) {
                     if (file.exists()) {
+                        if (debugEnabled) {
+                            LOGGER.debug("Deleting stale output file '{}'.", file.getAbsolutePath());
+                        }
```

Comment:
You already have the `absolutePath` above, so no need to get it from the file. Also, the `file`'s `path` is an absolute path, so `File.toString()` will also return the absolute path.

In either case there's no need for `debugEnabled`.

Reply:
You are right. I removed the field and the `if`.